backend/server.js

```diff
@@ -2,28 +2,95 @@ import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
+import crypto from 'crypto'
+import bcrypt from 'bcrypt'
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/authAPI"
+const mongoUrl = process.env.MONGO_URL || 'mongodb://localhost/authAPI'
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
+const User = mongoose.model('User', {
+  name: {
+    type: String, 
+    unique: true,
+    required: true,
+  },
+  email: {
+    type: String,
+    unique: true,
+    required: true,
+  },
+  password: {
+    type: String,
+    required: true,
+  },
+  accessToken: {
+    type: String,
+    default: () => crypto.randomBytes(128).toString('hex'),
+  },
+})
+
+const authenticateUser = async (req, res, next) => {
+  try {
+    const user = await User.findOne({ accessToken: req.header('Authorization') })
+    if (user) {
+      req.user = user
+      next()
+    } else {
+      res.status(401).json({ loggedOut: true, message: 'Please try logging in again' })
+    }
+  } catch (err) {
+    res.status(403).json({ message: 'Access token is missing or wrong', errors: err })
+  }
+}
 //   PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
 // Start defining your routes here
+
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: "Service unavailable" })
+  }
+})
+
 app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
-// Start the server
+app.post('/users', async (req, res) => {
+  try {
+    const {name, email, password} = req.body
+    const SALT = bcrypt.genSaltSync(10)
+    const user = new User({name, email, password: bcrypt.hashSync(password, SALT)})
+    await user.save()
+    res.status(201).json({ userId: user._id, accessToken: user.accessToken })
+  } catch (err) {
+    res.status(400).json({ message: 'Could not create user', errors: err.errors })
+  }
+})
+
+app.get('/secrets', authenticateUser)
+app.get('/secrets', async (req, res) => {
+  const secretMessage = `This is a super secret message for ${req.user.name}`
```

Comment:
When you login and reveal the secret message it is showing like this: {"secretMessage":"This is a super secret message for test"}
To fix this you need to wrap your template string in {} like this: const secretMessage = {`This is a super secret message for ${req.user.name}`}